source/brailleDisplayDrivers/superBraille.py

```diff
@@ -0,0 +1,142 @@
+#A part of NonVisual Desktop Access (NVDA)
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+#Copyright (C) 2017 NV Access Limited, Coscell Kao
+
+import serial
+from collections import OrderedDict
+import braille
+import hwPortUtils
+import hwIo
+import time
+import inputCore
+from logHandler import log
+
+BAUD_RATE = 9600
+TIMEOUT = 0.5
+
+# Tags sent by the SuperBraille
+# Sent to identify the display and receive amount of cells this unit has
+DESCRIBE_TAG = "\xff\xff\x0a"
+# Sent to request displaying of cells
+DISPLAY_TAG = "\xff\xff\x04\x00\x99\x00\x50\x00"
+
+class BrailleDisplayDriver(braille.BrailleDisplayDriver):
+	name = "superBrl"
+	# Translators: Names of braille displays.
+	description = _("SuperBraille")
+	isThreadSafe=True
+	_dev=None
+
+	USB_IDs = {
+		"USB\\VID_10C4&PID_EA60", # SuperBraille 3.2
+	}
+
+	@classmethod
+	def getPossiblePorts(cls):
+		ports = OrderedDict()
+		comPorts = list(hwPortUtils.listComPorts(onlyAvailable=True))
+		try:
+			next(cls._getAutoPorts(comPorts))
+			ports.update((cls.AUTOMATIC_PORT,))
+		except StopIteration:
+			pass
+		for portInfo in comPorts:
+			# Translators: Name of a serial communications port.
+			ports[portInfo["port"]] = _("Serial: {portName}").format(portName=portInfo["friendlyName"])
+		print "ports: %s"%ports
+		return ports
+
+	@classmethod
+	def _getAutoPorts(cls, comPorts):
+		for portInfo in comPorts:
+			port = portInfo["port"]
+			hwID = portInfo["hardwareID"]
+			if any(hwID.startswith(x) for x in cls.USB_IDs):
+				portType = "USB serial"
+			else:
+				continue
+			yield port, portType
+
+	@classmethod
+	def check(cls):
+		return True
+
+	def __init__(self,port="Auto"):
+		super(BrailleDisplayDriver, self).__init__()
+		found = False
+		if port == "auto":
+			tryPorts = self._getAutoPorts(hwPortUtils.listComPorts(onlyAvailable=True))
+		else:
+			tryPorts = ((port, "serial"),)
+		for port, portType in tryPorts:
+			log.debug("Checking port %s for a SuperBraille", port)
+			try:
+				self._dev = hwIo.Serial(port, baudrate=BAUD_RATE, stopbits=serial.STOPBITS_ONE, parity=serial.PARITY_NONE, timeout=TIMEOUT, writeTimeout=TIMEOUT, onReceive=self._onReceive)
+				log.debug("Port opened.")

```

Comment:
I don't think these debug messages are needed because we can get this info (and more) from the hwIo debug logging.